src/main/java/training/chessington/model/pieces/Knight.java

```diff
@@ -15,6 +16,40 @@ public Knight(PlayerColour colour) {
 
     @Override
     public List<Move> getAllowedMoves(Coordinates from, Board board) {
-        return new ArrayList<>();
+        List<Move> allowableMoves = new ArrayList<>();
+
+        // laying rectangle
+        getCornersOfRectangle(from, board, allowableMoves, 1, 2);
+        // standing rectangle
+        getCornersOfRectangle(from, board, allowableMoves, 2, 1);
+
+        return allowableMoves;
+    }
+
+    private void getCornersOfRectangle(Coordinates from, Board board,List<Move> initAllowableMoves, int rowInc, int colInc) {
+        int fromRow = from.getRow();

```

Comment:
Couple things about this method:

- Normally a method that starts with "get" should return something
- `rowInc` and `colInc` aren't very descriptive, it could mean inclusive or increment? I'd recommend explicitly naming them

The method name doesn't really explain much to me. I can't tell just from the name what exactly this is trying to do. After reading through the method I _think_ it's getting a 2x3 rectangle for the different potential knight positions (and each call returns 4 out of 8 potential positions). It's quite difficult to understand exactly how this is achieved and so I'm not sure its better than just hard coding the 8 different positions for a knight, at least that way its very clear exactly what the code is doing.

What do you think? :D